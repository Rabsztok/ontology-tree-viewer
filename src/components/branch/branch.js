import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  justify-content: stretch;
  border-bottom: ${props => props.theme.border};
  width: 100%;
  
  &:nth-last-child(1) {
    border-bottom: none;
  }
`

const Name = styled.div`
  padding: ${props => props.theme.cellPadding};
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  border-right: ${props => props.theme.border};
  flex-grow: 0;
  background: ${props => props.active ? props.theme.activeColor : 'none'};
  width: 35%;
  transition: background .4s;
`

const Children = styled.div`
  display: flex;
  flex-direction: column;
  flex-grow: 0;
  justify-content: stretch;
  width: 100%;
  border-right: 1px solid #ddd;
`

const Branch = ({leaf, name, children, active, onClick}) => {
  if (active === undefined) console.log(name, leaf)
  return (
    <Wrapper onClick={onClick}>
      <Name active={active} leaf={leaf}>{name}</Name>
      <Children>{children}</Children>
    </Wrapper>
  )
}

Branch.propTypes = {
  leaf: PropTypes.bool,
  name: PropTypes.string,
  children: PropTypes.array,
  active: PropTypes.bool.isRequired,
  onClick: PropTypes.func
}


export { Branch }
export default Branch