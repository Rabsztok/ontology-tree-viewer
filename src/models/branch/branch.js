import { types } from 'mobx-state-tree'

const inheritanceWarning = {
  type: 'InheritanceError',
  message:
    'Model branch is abstractive. Inherit from it and override `children` getter'
}

/** Abstractive class other branch-based models inherits from it */
const Branch = types
  .model('Branch', {
    id: types.identifierNumber,
    name: types.string
  })
  .views(self => ({
    get active () {
      return self.children.some(child => child.active)
    },
    get children () {
      throw inheritanceWarning
    }
  }))

export default Branch
