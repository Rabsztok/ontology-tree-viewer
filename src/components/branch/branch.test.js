import React from 'react'
import { shallow } from 'enzyme'
import { Branch } from './branch'

describe('Branch', () => {
  it('renders without errors', () => {
    const component = shallow(
      <Branch name="Branch" active={false}/>
    )
    expect(component).toMatchSnapshot()
  })
})