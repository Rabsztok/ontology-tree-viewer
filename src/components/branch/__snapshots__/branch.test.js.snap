// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branch renders without errors 1`] = `
<styled.div>
  <styled.div
    active={false}
  >
    Branch
  </styled.div>
  <styled.div />
</styled.div>
`;
